posthog/management/commands/sync_persons_to_clickhouse.py

```diff
@@ -0,0 +1,151 @@
+        Recommended: run first without `--live-run` and first for person table, then distinct_id table
+        """
+
+    def add_arguments(self, parser):
+        parser.add_argument("--team-id", default=None, type=int, help="Specify a team to fix data for.")
+        parser.add_argument(
+            "--tables", default=[PERSON_TABLE, DISTINCT_ID_TABLE], action="append", help="Which tables to sync"
+        )
+        parser.add_argument(
+            "--deletes", action="store_true", help="process deletes for data in ClickHouse but not Postgres"
+        )
+        parser.add_argument("--live-run", action="store_true", help="Run changes, default is dry-run")
+
+    def handle(self, *args, **options):
+        run(options)
+
+
+def run(options):
+    live_run = options["live_run"]
+    deletes = options["deletes"]
+
+    if not options["team_id"]:
+        logger.error("You must specify --team-id to run this script")
+        exit(1)
+
+    if not set(options["tables"]) <= {PERSON_TABLE, DISTINCT_ID_TABLE}:
+        logger.error("Illegal value in --tables")
+        exit(1)
+
+    team_id = options["team_id"]
+
+    if PERSON_TABLE in options["tables"]:
+        run_person_sync(team_id, live_run, deletes)
+
+    if DISTINCT_ID_TABLE in options["tables"]:
+        run_distinct_id_sync(team_id, live_run, deletes)
+
+
+def run_person_sync(team_id: int, live_run: bool, deletes: bool):
+    logger.info("Running person table sync")
+    # lookup what needs to be updated in ClickHouse and send kafka messages for only those
+    persons = Person.objects.filter(team_id=team_id)
+    rows = sync_execute(
+        """
+            SELECT id, max(version) FROM person WHERE team_id = %(team)s GROUP BY id HAVING max(is_deleted) = 0
+        """,
+        {
+            "team": team_id,
+        },
+    )
+    ch_persons_to_version = {row[0]: row[1] for row in rows}
+
+    for person in persons:
+        ch_version = ch_persons_to_version.get(person.uuid, None)
+        if ch_version > person.version:
+            logger.info(
+                f"Clickhouse version ({ch_version}) for '{person.uuid}' is higher than in Postgres ({person.version}). Ignoring."
+            )
+            continue
+        if ch_version != person.version:
+            logger.info(f"Updating {person.uuid} to version {person.version}")
+            if live_run:
+                # Update ClickHouse via Kafka message
+                create_person(
+                    team_id=team_id,
+                    version=person.version,
+                    uuid=str(person.uuid),
+                    properties=person.properties,  # TODO: check that the format is correct
+                    is_identified=person.is_identified,
+                    created_at=person.created_at,  # TODO: check formatting
+                )
+
+    if deletes:
+        logger.info("Processing person deletions")
+        postgres_uuids = {person.uuid for person in persons}
```

Comment:
non-blocker but should we be iterating over `persons` twice? there could be **a lot** of records.

maybe we can delete records from the CH dict on the first loop and then just handle the remaining CH ids on the delete portion

Reply:
I think it's fine to iterate twice - it's not going to be something we're running a lot. The code is simpler / more fool proof this way